catalyst/core/runner.py

```diff
@@ -350,8 +351,24 @@ def _run_stage(self, stage: str):
                 break
 
             self.state.epoch += 1
+            self.state.stage_epoch += 1
         self._run_event("stage", moment="end")
 
+    def get_start_stage(self):
+        state_params = self.experiment.get_state_params(
```

Comment:
maybe, it's better to add something like `resume` to Experiment to prevent all this hack around stages and state?

Reply:
I think it should be placed in `Runner`, because `Runner` has full information about current learning process, `Runner` emit events and decide, which `stage` will calculate. In first vestion `get_start_stage` was placed after `_prepare_for_stage`. But in this case it looks like a hack, because you need to write something like this:
```
def _run_stage(self, stage: str):
        self._prepare_for_stage(stage)
        resume_stage = self.get_resume_stage()
        stages = list(self.experiment.stages)
        if stages.index(resume_stage) > stages.index(stage):
            return
```

Reply:
as the Runner is mostly about *how* to run the experiment, and experiments is mostly about *what* to run, I think it looks okay to place initial `resume` into experiment. 
maybe, with @gazay abstraction for ConfigParser, that will create appropriate Experiment.Config from any user config this hypothesis will also work too

Reply:
As for me saving and restoring is the part of _how_, because Runner can compute different stages with full restarting on each epoch. Now `resume` placed in `checkpoint` callback. As for me, using callback for resuming looks like a hack. Changes in this PR  focused on minimal changes, which will fix the bug. But now it looks like refactoring of architecture. 

I think resuming can be implemented in Experiment constructor, which returns cutted list of stages.  This approach is more intuitive. But in this case, restoring must be removed from checkpointcallback.

Reply:
yes, maybe checkpoint callback for restoing should be implemented in another logic, 
nevertheless, it the story for another PR
for now I think, that start stage logic should be on the Experiment side

Reply:
I move get_start_stage from Runner to ConfigExperiment.